sea-orm-macros/src/derives/value_type.rs

```diff
@@ -0,0 +1,239 @@
+                    "DateTime" | "NaiveDateTime" => {
+                        quote! { ChronoDateTime }
+                    }
+                    "DateTimeUtc" | "DateTimeLocal" | "DateTimeWithTimeZone" => {
+                        quote! { ChronoDateTimeWithTimeZone }
+                    }
+                    "Uuid" => quote! { Uuid },
+                    "Json" => quote! { Json },
+                    "Decimal" => quote! { Decimal },
+                    _ => {
+                        // Assumed it's ActiveEnum if none of the above type matches
+                        quote! {}
+                    }
+                };
+                if arr_type.is_empty() {
+                    let field_span = field.span();
+                    let ty: Type = LitStr::new(field_type, field_span).parse().expect("g");
+                    let def = quote_spanned! { field_span =>
+                        std::convert::Into::<sea_orm::ArrayType>::into(
+                            <#ty as sea_orm::sea_query::ValueType>::array_type()
+                        )
+                    };
+                    quote! { #def }
+                } else {
+                    quote! { sea_orm::sea_query::ArrayType::#arr_type }
+                }
+            }
+        };
+
+        Some(DeriveValueType {
+            name,
+            ty,
+            column_type,
+            array_type,
+        })
+    }
+
+    fn expand(&self) -> syn::Result<TokenStream> {
+        let expanded_impl_value_type: TokenStream = self.impl_value_type();
+        Ok(expanded_impl_value_type)
+    }
+
+    fn impl_value_type(&self) -> TokenStream {
+        let name = &self.name;
+        let ty = &self.ty;
+        let column_type = &self.column_type;
+        let array_type = &self.array_type;
+
+        quote!(
+            #[automatically_derived]
+            impl From<#name> for Value {
+                fn from(source: #name) -> Self {
+                    source.0.into()
+                }
+            }
+
+            #[automatically_derived]
+            impl sea_orm::TryGetable for #name {
+                fn try_get_by<I: sea_orm::ColIdx>(res: &QueryResult, idx: I) -> Result<Self, sea_orm::TryGetError> {
+                    <#ty as sea_orm::TryGetable>::try_get_by(res, idx).map(|v| #name(v))
+                }
+            }
+
+            #[automatically_derived]
+            impl sea_query::ValueType for #name {
+                fn try_from(v: Value) -> Result<Self, sea_query::ValueTypeErr> {
+                    <#ty as sea_query::ValueType>::try_from(v).map(|v| #name(v))
+                }
+
+                fn type_name() -> String {
+                    stringify!(#name).to_owned()
+                }
+
+                fn array_type() -> sea_orm::sea_query::ArrayType {
+                    #array_type
+                }
+
+                fn column_type() -> sea_orm::sea_query::ColumnType {
+                    #column_type
+                }
+            }
```

Comment:
Please change the `sea_query::xxx` namespace to `sea_orm::sea_query::xxx`. User might not have `sea_query` crate on the root.